jekyll/_cci2/test-splitting-tutorial.adoc

```diff
@@ -0,0 +1,248 @@
+
+* You should also now see five **parallel runs**, as a result of the number of execution environments set by the `parallelism` key. Each Docker environment is labeled by its index number (so you have numbers 0 through 4). You can click on each environment or node to see the individual steps in each parallel run. The environment you are viewing will be highlighted in green.
++
+image::{{site.baseurl}}/assets/img/docs/test-splitting-parallel-runs.png[Five parallel runs with run times displayed]
++
+You might also notice that the parallel run times are not all equal, nor is the overall run time of the pipeline cut down to precisely 1/5. Each executor runs the same steps, but there is a difference in terms of which environment runs which tests. There may also be some variation in how long each executor takes to spin up. 
++
+Splitting tests by timing is the best way to ensure tests are split as evenly as possible and parallel runs finish around the same time. With that said, you may need to play around with the parallelism level to find the number that works best for you.
+
+* In any of the parallel runs, open the **Test application** step. You'll see which test suites and how many individual tests were executed in this particular run. You'll also see this message in the output:
++
+    Error reading historical timing data: file does not exist
+    Requested weighting by historical based timing, but they are not present. Falling back to weighting by name.
++
+Since this is the first time you are storing test data from the pipeline, CircleCI does not currently have timing data to work with, so it defaults to splitting tests by name. 
+
+* Open the **Timing** tab in the job. This tab provides a visualization of how each parallel run did relative to each other.
++
+image::{{site.baseurl}}/assets/img/docs/test-splitting-timing-tab.png[Parallel runs visualization in Timings tab]
++
+The chart indicates which three steps within each run took the longest to complete. Hover over each section of the bar to see those respective steps.
++
+You may also notice on the upper right corner within the Timing tab an indicator for idle time. In this pipeline, there was a total of 11 seconds between each finished run and the end of the longest run.
+
+== Step four - Trigger pipeline that uses test timing data
+
+In the previous step, you saw that test splitting defaulted to splitting tests based on name. Now that test data has been saved, CircleCI can now split your tests by timing the next time the pipeline runs.
+
+. Commit a change in your project to trigger the pipeline again. 
++
+For example, you can try upgrading to a newer version of the Node orb, such as `circleci/node@5.0.2`. Or, you may choose to just trigger a pipeline again, by clicking the Rerun button in the web app and selecting **Rerun Workflow from Start**. 
```

Comment:
I tried rerunning the workflow, and triggering a pipeline from the app and each time it still said theres no timing data so defaulted to using name-based splitting

Reply:
Same for committing a change to the config file. I wonder why?!

Reply:
I've noticed that whenever I'm trying to commit the last config change using the in-app config editor it's failing. I'm going to try pulling and completing it locally instead.

Reply:
Seems to be working now (I feel like there might be a bug using the in-app config editor)

Reply:
A note on the wording here though - You need to point them to the  "Trigger a Pipeline" option in app, rather than rerun a workflow, because rerunning a workflow would happen within the same pipeline so not sure that would change the outcome of the timing data etc

![Screenshot 2022-05-10 at 13 25 23](https://user-images.githubusercontent.com/24589403/167627742-964b0962-79f2-481c-bd2f-59c38fcc7a31.png)
.

Reply:
It looks like I tried the Rerun option which seemed to have worked... but I think maybe going with Trigger Pipeline is better.